File: bindings/nodejs/examples/how_tos/client/get-health.ts
// Copyright 2021-2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, initLogger } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/client/get-health.ts

// In this example we will get the node health
async function run() {
    initLogger();
    for (const envVar of ['NODE_URL']) {
        if (!(envVar in process.env)) {
            throw new Error(`.env ${envVar} is undefined, see .env.example`);
        }
    }

    const client = new Client({
        // Insert your node URL in the .env.
        nodes: [process.env.NODE_URL as string],
        localPow: true,
    });

    try {
        const isHealthy = await client.getHealth(
            process.env.NODE_URL as string,
        );
        console.log('Healthy: ', isHealthy);
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());

File: bindings/python/examples/how_tos/client/get_health.py
import os

from dotenv import load_dotenv

from iota_sdk import Client

load_dotenv()

node_url = os.environ.get('NODE_URL', 'https://api.testnet.shimmer.network')

# Create a Client instance
client = Client(nodes=[node_url])

# Get the node health
is_healthy = client.get_health(node_url)
print(f'Healthy: {is_healthy}')
